modules/core/src/main/java/org/apache/ignite/internal/client/thin/ClientServicesImpl.java

```diff
@@ -281,6 +344,96 @@ else if (ch.clientChannel().protocolCtx().isFeatureSupported(ProtocolBitmaskFeat
         }
     }
 
+    /**
+     * Asynchronously requests the service topology if the partition awareness is enabled and if the update is requred.
+     *
+     * @param srvcTop If not {@code null}, uses this service topology record and ignores {@code name}.
+     * @param name If {@code srvTop} is {@code null}, gets service topology record by this name.
+     * @see #needUpdateSrvcTop(ServiceTopology, AffinityTopologyVersion)
+     */
+    private void tryRequestServiceTopology(@Nullable ServiceTopology srvcTop, String name) {
+        if (srvcTop == null) {
+            srvcTop = servicesTopologies.compute(name, (nm, t) -> {
+                if (t == null)
+                    t = new ServiceTopology();
+
+                return t;
+            });
+        }
+
+        ServiceTopology srvcTop0 = srvcTop;
+
+        AffinityTopologyVersion curAffTop = ch.affinityContext().lastTopology().version();
+
+        if (!needUpdateSrvcTop(srvcTop0, curAffTop))
+            return;
+
+        ForkJoinPool.commonPool().execute(() -> {
+            Throwable t = null;
+
+            try {
+                if (log.isDebugEnabled())
+                    log.debug("Requesting service topology update for the service '" + name + "' ...");
+
+                List<UUID> nodes = ch.service(
```

Comment:
Let's use `serviceAsync`, so we don't need a separate thread.

Reply:
Fixed